.github/workflows/self_hosted_e2e_docker.yaml

```diff
@@ -0,0 +1,93 @@
+# ------------------------------------------------------------
+# Copyright 2021 The Dapr Authors
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------
+
+name: E2E - Self-hosted-docker
```

Comment:
why are we having a new workflow created here? If it is for testing docker hub integration then why can't that be included potentially through the matrix definition ? 
https://docs.github.com/en/actions/using-jobs/using-a-build-matrix-for-your-jobs#using-environment-variables-in-a-matrix

Reply:
We should avoid multiple workflows for the same test at any cost.